tests/core/post_process/test_local_search.py

```diff
@@ -0,0 +1,131 @@
+    ising_matrix.to_ising_matrix(qaoa_encoded)
+    size = qaoa_encoded.num_bits()
+    state = np.random.choice([-1, 1], size=size)
+    random_index = np.random.randint(0, size)
+    delta_E = ising_matrix.calc_E_diff(state, random_index)
+
+    assert isinstance(delta_E, float)
+
+
+def test_run(local_search_instance, qaoa_converter):
+    size = qaoa_converter.ising_encode().num_bits()
+    state = np.random.choice([-1, 1], size=size)
+
+    invalid_method = "non_existent_method"
+    with pytest.raises(ValueError):
+        local_search_instance.run(state, local_search_method=invalid_method)
+
+
+def test_run_local_search(local_search_instance, qaoa_converter):
+    size = qaoa_converter.ising_encode().num_bits()
+    state = np.random.choice([-1, 1], size=size)
+
+    mock_method = MagicMock()
+
+    def side_effect(_, current_state, __):
+        if mock_method.call_count < 5:
+            new_state = current_state.copy()
+            new_state[0] *= -1
+        else:
+            new_state = current_state.copy()
+        return new_state
+
+    mock_method.side_effect = side_effect
+    final_state = local_search_instance._run_local_search(
+        mock_method, state, max_iter=10
+    )
+
+    assert mock_method.call_count == 5
+    assert len(state) == len(final_state)
+    assert all(val in [-1, 1] for val in final_state)
+
+    mock_method.reset_mock()
+    final_state_2 = local_search_instance._run_local_search(
+        mock_method, state, max_iter=3
+    )
+    assert mock_method.call_count == 3
+    assert len(state) == len(final_state_2)
+    assert all(val in [-1, 1] for val in final_state_2)
+
+
+def test_best_improvement(local_search_instance, ising_matrix, qaoa_converter):
+    qaoa_encoded = qaoa_converter.ising_encode()
+    ising_matrix.to_ising_matrix(qaoa_encoded)
+    size = qaoa_encoded.num_bits()
+    state = np.random.choice([-1, 1], size=size)
+    E1 = qaoa_encoded.calc_energy(state)
+
+    new_state = local_search_instance.best_improvement(ising_matrix, state, size)
+    E2 = qaoa_encoded.calc_energy(new_state)
+
+    assert len(state) == len(new_state)
+    assert all(val in [-1, 1] for val in new_state)
+    assert E2 <= E1
+
+
+def test_first_improvement(local_search_instance, ising_matrix, qaoa_converter):
+    qaoa_encoded = qaoa_converter.ising_encode()
+    ising_matrix.to_ising_matrix(qaoa_encoded)
+    size = qaoa_encoded.num_bits()
+    state = np.random.choice([-1, 1], size=size)
+    E1 = qaoa_encoded.calc_energy(state)
+
+    new_state = local_search_instance.first_improvement(ising_matrix, state, size)
+    E2 = qaoa_encoded.calc_energy(new_state)
+
+    assert len(state) == len(new_state)
+    assert all(val in [-1, 1] for val in new_state)
+    assert E2 <= E1
+
+
+def test_decode(local_search_instance, qaoa_converter):
+    size = qaoa_converter.ising_encode().num_bits()
+    state = np.random.choice([-1, 1], size=size)
+    sampleset = local_search_instance.decode(state)
```

Comment:
Check the original state and expected state in the sampleset is same or not and objective value is what you expect or not